misc/scripts/check_ci_log.py

```diff
@@ -43,23 +46,75 @@
+# Example - https://github.com/godotengine/godot/issues/37980#issue-602414919
 
 if file_contents.find("Assertion failed") != -1:
-    print("ERROR: Assertion failed in project, check execution log for more info")
+    print(
+        'ERROR: Assertion failed in project, check execution log for more info (search for "Assertion failed" in CI log)'
+    )
     sys.exit(55)
 
+# Sometimes pointers points at objects of different types, which may cause
+# to show errors like "runtime error: member call on address 0x1 which does not point to an object of type"
+# Example - https://github.com/godotengine/godot/issues/51351#issue-963170661
+
+# Waiting for https://github.com/godotengine/godot/issues/51888
+
+if file_contents.find("vptr for") != -1:
+    print('WARNING: Found pointer which not point at valid object (search for "vptr for" in CI log)')
+    # sys.exit(56)
+
+# By default overflow or underflow of signed values are in C++ just
+# undefined behavior(most of the time value is wrapped)
+# Example - https://github.com/godotengine/godot/issues/33644#issue-523623547
+
+if file_contents.find("cannot be represented in type") != -1 or file_contents.find("is outside the range") != -1:
+    print(
+        'ERROR: Found pointer which not point at valid object (search for "cannot be represented in type" or "is outside the range" in CI log)'
+    )
+    sys.exit(57)
+
+# Some functions like memcpy doesn't expect that its argument is null pointer.
+# This may later be cause of bugs or crashes.
+# Example - https://github.com/godotengine/godot/issues/48215#issue-867854743
+
+if file_contents.find("null pointer passed as argument") != -1:
+    print(
+        'ERROR: Found null pointer passed as argument to function which not expect it (search for "null pointer passed as argument" in CI log)'
+    )
+    sys.exit(58)
+
+# Casting or pointer moving caused that code trying to violate alignement rules
+# Example - https://github.com/godotengine/godot/issues/31203#issue-478487290
+
+if file_contents.find("misaligned address") != -1:
+    print('ERROR: Found usage of misaligned pointer (search for "misaligned address" in CI log)')
+    sys.exit(59)
+
 # For now Godot leaks a lot of rendering stuff so for now we just show info
 # about it and this needs to be re-enabled after fixing this memory leaks.
+# Example - https://github.com/godotengine/godot/issues/47941#issue-859356605
+
+# Blocked by https://github.com/godotengine/godot/issues/46833
 
 if file_contents.find("were leaked") != -1 or file_contents.find("were never freed") != -1:
-    print("WARNING: Memory leak was found")
+    print('WARNING: Memory leak was found (search for "were leaked" or "were never freed" in CI log)')
+
+# Usually error about trying to free invalid ID is caused by removing wrong object
+# Example - https://github.com/godotengine/godot/issues/49623#issue-921610423
+
+# Blocked by Swiftshader bugs in CI
+
+if file_contents.find("Attempted to free invalid ID") != -1:
```

Comment:
I think this is going to trigger some bugs as the rendering code is not entirely stable yet.

Reply:
I know that this is unstable so for now it only prints warning message and CI still works fine when this error appear.